main.go

```diff
@@ -43,12 +45,28 @@ type options struct {
 }
 
 func main() {
-	if err := run(); err != nil {
+	cxt, cancel := context.WithCancel(context.Background())
```

Comment:
Prow for a while was installing its own handlers.  Is this overriding those or in parallel?

Reply:
I believe this would be in parallel.  These particular changes stem from a conversation started over in #185 where we were called out for not passing a legitimate context into the prow client when updating a prowjob. I figured I would lay the ground work with this PR and then disseminate the context later.